python/ray/tune/examples/dragonfly_example.py

```diff
@@ -0,0 +1,46 @@
+"""This test checks that Dragonfly is functional.
+
+It also checks that it is usable with a separate scheduler.
+"""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import ray
+from ray.tune import run
+from ray.tune.schedulers import AsyncHyperBandScheduler
+from ray.tune.suggest.dragonfly import DragonflySearch
+
+if __name__ == "__main__":
+    import argparse
+    from dragonfly.opt.gp_bandit import EuclideanGPBandit
+    from dragonfly.exd.worker_manager import SyntheticWorkerManager
+    from dragonfly.utils.euclidean_synthetic_functions import get_syn_func_caller  # noqa: E501
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--smoke-test", action="store_true", help="Finish quickly for testing")
+    args, _ = parser.parse_known_args()
+    ray.init()
+
+    config = {
+        "num_samples": 10 if args.smoke_test else 50,
+        "config": {
+            "iterations": 100,
+        },
+        "stop": {
+            "timesteps_total": 100
+        },
+    }
+    func_caller = get_syn_func_caller(
+        "hartmann6", noise_type="gauss", noise_scale=0.1)
+    worker_manager = SyntheticWorkerManager(1, time_distro="const")
```

Comment:
Can you make it so that you don't actually need "func_caller" and "worker_manager"? 

This example ideally should only need to pass in a `func` to `run`. I haven't followed the other PR; why does `GPbandit` need the function?

Reply:
Hmm `run` does take in a `func` but yes it is from `func_caller`. As for whether we don't need `func_caller` and `worker_manager`, I'm afraid not since to initialize a `GPBandit` these two things are required. We could also ask Samy about this

Reply:
Should be able to change that requirement for GPBandit.